### PR TITLE
refactor(ListItem): createListItemPartitions

### PR DESCRIPTION
**Pull request by lukelooker**
`createListItemPartitions` accepts all `ListItemProps` and outputs the core pieces that make up `ListItem` (i.e. the item, the label, the detail). It outputs these pieces in a 2-tuple, the first element being a fragment that contains all elements to go "inside" the main container, and a second element being all elements that go "outside" the main container (which is just the detail when accessory mode is enabled).

The PR does the following:
- Adds `createListItemPartitions`
- Refactors `ListItem` pieces to utilize `createListItemPartitions`
- Deprecates `padding` on `ListItem` `detail`, now is always `false`

## Developer Checklist [ℹ️](https://github.com/looker-open-source/components/blob/main//CONTRIBUTING.md#developer-checklist)

- [ ] ♿️ Accessibility addressed
- [ ] 🌏 Localization & Internationalization addressed
- [ ] 🖼 Image Snapshot coverage
- [ ] 📚 Documentation updated
- [ ] 🧳 Bundle size impact addressed
- [ ] 🏁 Performance impacts addressed
- [ ] 👾 Browsers tested
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] IE11

### CODE REVIEW

File: packages/components/src/List/utils/listItemPadding.ts [hunk 1]
```diff
@@ -24,27 +24,24 @@
 
  */
 
-import { SpaceRamp } from '@looker/design-tokens'
+import { SpaceRamp } from 'packages/design-tokens/src/system'
```
Comment (lukelooker) [inline packages/components/src/List/utils/listItemPadding.ts:27]: Oops. Bad path here.

File: packages/components/src/List/ListItemDetail.tsx [hunk 2]
```diff
@@ -29,11 +29,20 @@ import React from 'react'
...
-  <HoverDisclosure width={width} visible={!hoverDisclosure}>
-    <div {...props} />
-  </HoverDisclosure>
-))<DetailOptions>`
+export type ListItemDetailProps = Omit<
+  DetailOptions,
+  'accessory' | 'padding'
+> & {
+  className?: string
+}
```
Comment (lukelooker) [inline packages/components/src/List/ListItemDetail.tsx]: Was `className` actually needed here? It seems like that will still pass through even though Typescript isn't aware of it.

File: packages/components/src/List/ListItemContent.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  }
+
+const ListItemContentInternal: FC<ListItemContentProps> = ({
+  children,
+  itemRole = 'button',
+  ...props
+}) => {
+  let Component = (p: CompatibleHTMLProps<HTMLElement>) => (
+    <Button {...p} type="button" />
+  )
```
Comment (ElliotTheHuman) [inline packages/components/src/List/ListItemContent.tsx:78]: Something about this line makes `TreeCollection` and some of the `Menu` tests really unhappy.

File: packages/components/src/Tree/Tree.tsx [hunk 4]
```diff
@@ -197,29 +199,40 @@ const TreeLayout = ({
...
+    ...accordionProps,
+  })
+
+  const innerAccordion = (
+    <TreeAccordion className="tree-accordion-disclosure">
+      {!partialRender && (
+        <Accordion2Disclosure {...domProps} {...disclosureProps} />
+      )}
+      {treeContent}
+    </TreeAccordion>
```
Comment (ElliotTheHuman) [inline packages/components/src/Tree/Tree.tsx:235]: It's incredible how easy it was to compose everything myself. `useAccordion2` already seems like a big win.

**Review by lukelooker — COMMENTED** [review]
Found 1 remaining oddity in manual testing. 

 1. Go to "Tree: basic" story
 2. Click the label text of a `Tree` (not a `TreeItem`)
 3. Focus doesn't land on the `Accordion` so `<enter>`, `up-arrow`, etc.. (keyboard navigation) doesn't function properly

`document.activeElement` shows that focus is landing on the `ListItemContent` instead of an accordion piece as would be expected (I think)

Anyway, happy to help poke at it more but wanted to write it down before I lost track.

**Review by ElliotTheHuman — APPROVED** [review]
LGTM!

**Pull request merged**